lib/i18n/js/formatters/js.rb

```diff
@@ -20,7 +25,7 @@ def header
         end
 
         def line(locale, translations)
-          json_literal = @pretty_print ? translations : %(JSON.parse('#{translations.gsub("'"){"\\'"}}'))
+          json_literal = @pretty_print ? translations : %(JSON.parse('#{translations.gsub(/'|\\/){|match| JSON_ESCAPE_MAP[match] }}'))
```

Comment:
Possible to use the keys of the constant hash?
Also use `Hash#fetch` to avoid returning `nil` silently
```suggestion
          json_literal = @pretty_print ? translations : %(JSON.parse('#{translations.gsub(/#{JSON_ESCAPE_MAP.keys.join("|")}/){|match| JSON_ESCAPE_MAP.fetch(match) }}'))
```

Reply:
Impossible to use `keys.join` because `too short escape sequence` error (maybe caused by unescaped key) so I use `Regexp.union`